[native] Switch to use string based arbitrator configs

tanjialiang opened the pull request:
## Description
As we move forward with adopting the new string based configurations for implementation specific arbitrator configs, we are removing the old configs and migrate the configs to the new shared arbitrator configs.

```
== RELEASE NOTES ==
```
Shared arbitrator related configs are renamed to the following:
shared-arbitrator.reserved-capacity
shared-arbitrator.memory-pool-initial-capacity
shared-arbitrator.global-arbitration-enabled
shared-arbitrator.memory-pool-reserved-capacity
shared-arbitrator.memory-pool-transfer-capacity
shared-arbitrator.memory-reclaim-max-wait-time
(new) shared-arbitrator.fast-exponential-growth-capacity-limit
(new) shared-arbitrator.slow-capacity-grow-pct
(new) shared-arbitrator.memory-pool-min-free-capacity
(new) shared-arbitrator.memory-pool-min-free-capacity-pct

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -113,41 +142,126 @@ The configuration properties of Presto C++ workers are described here, in alphab
 
-``query-reserved-memory-gb``
+``shared-arbitrator.reserved-capacity``
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
The formatting line (`^^^`) should have the same number of characters as the line above it.

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -113,41 +142,126 @@ The configuration properties of Presto C++ workers are described here, in alphab
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-* **Type:** ``boolean``
+``shared-arbitrator.memory-pool-initial-capacity``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
The formatting line (`^^^`) should have the same number of characters as the line above it.

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -113,41 +142,126 @@ The configuration properties of Presto C++ workers are described here, in alphab
+  The initial memory pool capacity in bytes allocated on creation.
+
+``shared-arbitrator.global-arbitration-enabled``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
The formatting line (^^^) should have the same number of characters as the line above it.

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -113,41 +142,126 @@ The configuration properties of Presto C++ workers are described here, in alphab
-``system-memory-gb``
-^^^^^^^^^^^^^^^^^^^^
+``shared-arbitrator.memory-pool-reserved-capacity``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -113,41 +142,126 @@ The configuration properties of Presto C++ workers are described here, in alphab
 
-  Set ``system-memory-gb`` to the available machine memory of the deployment.
+``shared-arbitrator.memory-pool-transfer-capacity``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -113,41 +142,126 @@ The configuration properties of Presto C++ workers are described here, in alphab
 
-  Number of drivers to use per task. Defaults to hardware CPUs.
+``shared-arbitrator.memory-reclaim-max-wait-time``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -113,41 +142,126 @@ The configuration properties of Presto C++ workers are described here, in alphab
+  zero, then there is no timeout.
+
+``shared-arbitrator.fast-exponential-growth-capacity-limit``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -113,41 +142,126 @@ The configuration properties of Presto C++ workers are described here, in alphab
+
+  When shared arbitrator grows memory pool's capacity, the growth bytes will
+  be adjusted in the following way:
+   - If 2 * current capacity is less than or equal to
```
`/Users/steveburnett/Documents/GitHub/presto/presto-docs/src/main/sphinx/presto_cpp/properties.rst:211: ERROR: Unexpected indentation.`


Adjust the indentation for this list until the gray vertical line in the screenshot does not appear in a local doc build. See [Building the documentation](https://github.com/prestodb/presto/tree/master/presto-docs#building-the-documentation) in the presto-docs README for information on local doc builds. 

<img width="806" alt="Screenshot 2024-08-22 at 2 47 20 PM" src="https://github.com/user-attachments/assets/bd621b9d-0a21-4ec2-8536-8ff78ef9e36d">

tanjialiang replied:
Thanks, just revised

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -113,41 +142,126 @@ The configuration properties of Presto C++ workers are described here, in alphab
+  disabled.
+
+``shared-arbitrator.slow-capacity-grow-pct``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -113,41 +142,126 @@ The configuration properties of Presto C++ workers are described here, in alphab
+  See description for 'shared-arbitrator.fast-exponential-growth-capacity-limit'
+
+``shared-arbitrator.memory-pool-min-free-capacity``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -113,41 +142,126 @@ The configuration properties of Presto C++ workers are described here, in alphab
+  'shared-arbitrator.memory-pool-min-free-capacity' are set.
+
+``shared-arbitrator.memory-pool-min-free-capacity-pct``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

steveburnett reviewed (changes requested):
Thanks for the doc! Just some nits of formatting, the content looks great.

xiaoxmeng reviewed (dismissed):
@tanjialiang LGTM. Thanks!

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -113,41 +142,128 @@ The configuration properties of Presto C++ workers are described here, in alphab
-  system, such as disk spilling and cache prefetch.
+  The amount of memory in bytes reserved for each query memory pool. When
+  a query tries to allocate memory from the reserved space whose size is
+  specified by 'shared-arbitrator.reserved-capacity', it cannot allocate
```
```suggestion
  specified by ``shared-arbitrator.reserved-capacity``, it cannot allocate
```

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -113,41 +142,128 @@ The configuration properties of Presto C++ workers are described here, in alphab
+  The amount of memory in bytes reserved for each query memory pool. When
+  a query tries to allocate memory from the reserved space whose size is
+  specified by 'shared-arbitrator.reserved-capacity', it cannot allocate
+  more than the value specified in 'shared-arbitrator.memory-pool-reserved-capacity'.
```
```suggestion
  more than the value specified in ``shared-arbitrator.memory-pool-reserved-capacity``.
```

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -113,41 +142,128 @@ The configuration properties of Presto C++ workers are described here, in alphab
+  be adjusted in the following way:
+
+  * If 2 * current capacity is less than or equal to
+    'shared-arbitrator.fast-exponential-growth-capacity-limit', grow
```
```suggestion
    ``shared-arbitrator.fast-exponential-growth-capacity-limit``, grow
```

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -113,41 +142,128 @@ The configuration properties of Presto C++ workers are described here, in alphab
+    through fast path by at least doubling the current capacity, when
+    conditions allow (see below NOTE section).
+  * If 2 * current capacity is greater than
+    'shared-arbitrator.fast-exponential-growth-capacity-limit', grow
```
```suggestion
    ``shared-arbitrator.fast-exponential-growth-capacity-limit``, grow
```

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -113,41 +142,128 @@ The configuration properties of Presto C++ workers are described here, in alphab
+  * If 2 * current capacity is greater than
+    'shared-arbitrator.fast-exponential-growth-capacity-limit', grow
+    through slow path by growing capacity by at least
+    'shared-arbitrator.slow-capacity-grow-pct' * current capacity if
```
```suggestion
    ``shared-arbitrator.slow-capacity-grow-pct`` * current capacity if
```

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -113,41 +142,128 @@ The configuration properties of Presto C++ workers are described here, in alphab
+  adjusted growth bytes will not be respected.
+
+  NOTE: Capacity growth adjust is only enabled if both
+  'shared-arbitrator.fast-exponential-growth-capacity-limit' and
```
```suggestion
  ``shared-arbitrator.fast-exponential-growth-capacity-limit`` and
```

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -113,41 +142,128 @@ The configuration properties of Presto C++ workers are described here, in alphab
+
+  NOTE: Capacity growth adjust is only enabled if both
+  'shared-arbitrator.fast-exponential-growth-capacity-limit' and
+  'shared-arbitrator.slow-capacity-grow-pct' are set, otherwise it is
```
```suggestion
  ``shared-arbitrator.slow-capacity-grow-pct`` are set, otherwise it is
```

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -113,41 +142,128 @@ The configuration properties of Presto C++ workers are described here, in alphab
+* **Type:** ``string``
+* **Default value:** ``0.25``
+
+  See description for 'shared-arbitrator.fast-exponential-growth-capacity-limit'
```
```suggestion
  See description for ``shared-arbitrator.fast-exponential-growth-capacity-limit``.
```

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -113,41 +142,128 @@ The configuration properties of Presto C++ workers are described here, in alphab
+  pool's current memory usage:
+
+  * Free capacity is greater or equal to capacity *
+    'shared-arbitrator.memory-pool-min-free-capacity-pct'
```
```suggestion
    ``shared-arbitrator.memory-pool-min-free-capacity-pct``
```

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -113,41 +142,128 @@ The configuration properties of Presto C++ workers are described here, in alphab
+  * Free capacity is greater or equal to capacity *
+    'shared-arbitrator.memory-pool-min-free-capacity-pct'
+  * Free capacity is greater or equal to
+    'shared-arbitrator.memory-pool-min-free-capacity'
```
```suggestion
    ``shared-arbitrator.memory-pool-min-free-capacity``
```

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -113,41 +142,128 @@ The configuration properties of Presto C++ workers are described here, in alphab
+    'shared-arbitrator.memory-pool-min-free-capacity'
+
+  NOTE: In the conditions when original requested shrink bytes ends up
+  with more free capacity than above 2 conditions, the adjusted shrink
```
```suggestion
  with more free capacity than above two conditions, the adjusted shrink
```

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -113,41 +142,128 @@ The configuration properties of Presto C++ workers are described here, in alphab
+  bytes is not respected.
+
+  NOTE: Capacity shrink adjustment is enabled when both
+  'shared-arbitrator.memory-pool-min-free-capacity-pct' and
```
```suggestion
  ``shared-arbitrator.memory-pool-min-free-capacity-pct`` and
```

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -113,41 +142,128 @@ The configuration properties of Presto C++ workers are described here, in alphab
+
+  NOTE: Capacity shrink adjustment is enabled when both
+  'shared-arbitrator.memory-pool-min-free-capacity-pct' and
+  'shared-arbitrator.memory-pool-min-free-capacity' are set.
```
```suggestion
  ``shared-arbitrator.memory-pool-min-free-capacity`` are set.
```

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -113,41 +142,128 @@ The configuration properties of Presto C++ workers are described here, in alphab
+* **Type:** ``string``
+* **Default value:** ``0.25``
 
+  See description for 'shared-arbitrator.memory-pool-min-free-capacity'
```
```suggestion
  See description for ``shared-arbitrator.memory-pool-min-free-capacity``.
```

steveburnett commented on presto-docs/src/main/sphinx/presto_cpp/properties.rst:
```diff
@@ -95,14 +95,43 @@ The configuration properties of Presto C++ workers are described here, in alphab
+* **Type:** ``integer``
+* **Default value:** ``40``
+
+  Memory allocation limit enforced via internal memory allocator. It consists of two parts:
```
```suggestion
  Memory allocation limit enforced by a internal memory allocator. It consists of two parts:
```
See the [Gitlab documentation style guide recommended word list entry for via](https://docs.gitlab.com/ee/development/documentation/styleguide/word_list.html#via) for discussion and alternate suggestions if my suggestion here doesn't work for you.

steveburnett reviewed (changes requested):
Thank you for the quick revision! I took another look and found some more nits of formatting and phrasing, but I think this should be everything.

tanjialiang commented:
> Thank you for the quick revision! I took another look and found some more nits of formatting and phrasing, but I think this should be everything.

Hi @steveburnett , just addressed your comments. Would you help to take another look? Thanks

steveburnett reviewed (approved):
LGTM! (docs) 

Pull updated branch, new doc build, looks good. Thanks!
